src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClient.h

```diff
@@ -29,26 +29,62 @@ namespace client
              typename AuthSchemesVariantT,
              typename EndpointProviderT,
              typename SerializerT,
-             typename ResponseT>
+             typename ResponseT,
+             typename MarshallerT>
     class AwsSmithyClientT : public AwsSmithyClientBase
     {
+    private: 
+        
     public:
+        static_assert(std::is_base_of<Aws::Client::AWSErrorMarshaller, MarshallerT>::value, "MarshallerT must be derived from class Aws::Client::AWSErrorMarshaller");
+
         explicit AwsSmithyClientT(const ServiceClientConfigurationT& clientConfig, const Aws::String& serviceName,
             const std::shared_ptr<Aws::Http::HttpClient>& httpClient,
             const std::shared_ptr<Aws::Client::AWSErrorMarshaller>& errorMarshaller,
             const std::shared_ptr<EndpointProviderT> endpointProvider,
             const std::shared_ptr<ServiceAuthSchemeResolverT>& authSchemeResolver,
             const Aws::UnorderedMap<Aws::String, AuthSchemesVariantT>& authSchemes)
-            : AwsSmithyClientBase(Aws::MakeUnique<ServiceClientConfigurationT>(ServiceNameT, clientConfig), serviceName, httpClient, errorMarshaller),
+            : AwsSmithyClientBase(Aws::MakeShared<ServiceClientConfigurationT>(ServiceNameT, clientConfig), serviceName, httpClient, errorMarshaller),
               m_clientConfiguration(*static_cast<ServiceClientConfigurationT*>(AwsSmithyClientBase::m_clientConfig.get())),
               m_endpointProvider(endpointProvider),
               m_authSchemeResolver(authSchemeResolver),
               m_authSchemes(authSchemes),
-              m_serializer(Aws::MakeUnique<SerializerT>(ServiceNameT, m_clientConfiguration.telemetryProvider))
+              m_serializer(Aws::MakeShared<SerializerT>(ServiceNameT, m_clientConfiguration.telemetryProvider))
         {
             m_serviceName = ServiceNameT;
         }
 
+        AwsSmithyClientT(const AwsSmithyClientT& other):
+            AwsSmithyClientBase(Aws::MakeShared<ServiceClientConfigurationT>(ServiceNameT, other.m_clientConfig), other.m_serviceName, Aws::Http::CreateHttpClient(*other.m_clientConfig), Aws::MakeShared<MarshallerT>(other.m_serviceName.c_str())),
```

Comment:
for the sake of consistency `MakeShared` should be called with the same log tag, i.e.
`Aws::MakeShared<ServiceClientConfigurationT>(ServiceNameT, other.m_clientConfig)`
vs
`Aws::MakeShared<MarshallerT>(other.m_serviceName.c_str())`

`other.m_serviceName` should be the same `ServiceNameT` but since this is the class that is calling `MakeShared` lets make it consistent to `ServiceNameT` for both